File: Backend/src/services/interfaces/IRolesServices.ts
import { Roles } from "../../domain/Entities/Roles";
import { RolesRepository } from "../repositories";
import { IServices } from "./IServices";

interface IRolesServices extends IServices<Roles> {
    GetRoleByRoleName(roleName: string): Promise<Roles>;
    VerifyRoleNameExists(roleName:string): Promise<boolean>
}
export { IRolesServices }

File: BackEnd/Services/Services/IServices/IRolesServices.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bussines.Services.IServices
{
    public interface IRolesServices
    {
        Task<string> Save(Roles roles);
        Task<string> Delete(Roles roles);
    }
}
